### ISSUE TITLE
Improve multi-tagging UX

### ISSUE THREAD

**Issue by udoprog** [nuttyartist/notes#354]
Currently when tagging a note the context menu which is used to tag items is a bit noisy (says `Remove tag <tag>` / `Add tag <tag>`) and gets collapsed as you add or remove individual tags leaving you to have to "chase" the menu again for each tag added or removed.

![image](https://user-images.githubusercontent.com/111092/190875771-f7468034-ad7a-429d-ae00-de6a7d5f1418.png)

An improvement to me would be if:
* The menu indicates which tags apply with a checkbox next to each tag (along with its color) that can be toggled by clicking and;
* The context menu isn't collapsed by a tag being toggled but rather dismissed by e.g. clicking outside of the menu area.

Cheers!

**Comment by nuttyartist** [comment]
> The context menu isn't collapsed by a tag being toggled but rather dismissed by e.g. clicking outside of the menu area.

This is very important, I had this on my mind but forgot to create an issue for that. Thanks!

> The menu indicates which tags apply with a checkbox next to each tag (along with its color) that can be toggled by clicking and;

What would this look like? Something like this?

![Screen Shot 2022-09-18 at 11 42 38 AM](https://user-images.githubusercontent.com/16375940/190891312-c3ecce28-edcc-4321-89ed-f4e05ac16543.png)

**Comment by udoprog** [comment]
Yeah, that looks great, esp. with the grouping. I'd also consider dropping the extra wording (and use only the tag name) in favour of a hint when hovering.

Thanks!
